api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -122,27 +122,30 @@
    *
    * @param conflictDetectionFilter an expression on rows in the table
    * @return this for method chaining
+   * @deprecated this will be removed in 0.14.0;
+   *             use {@link #validateNoConflictingOperations(Expression)} instead
```

Comment:
Do we also need to update `ReplacePartitions`?

Reply:
Let me check during the weekend

Reply:
I am not sure we would do anything differently in `ReplacePartitions` if anybody commits new delete files.
Any particular thoughts you had in mind, @rdblue?

Reply:
@szehon-ho is adding support for detecting conflicts in `ReplacePartitions`. The original behavior was basically snapshot isolation like Hive. But you could argue that it would be valuable to support serializable by ensuring the replaced partitions haven't changed since some starting snapshot. This is the PR: https://github.com/apache/iceberg/pull/2925

Reply:
I forgot about that PR. I'll take a look on Monday.